devsetup/Makefile

```diff
@@ -44,11 +45,20 @@ download_tools: ## Runs an ansible playbook to install required tools with the v
 	download_tools.yaml
 
 .PHONY: crc_attach_default_interface
-crc_attach_default_interface:
-	sudo virsh attach-interface crc --source default --type network --model virtio --config --persistent
+crc_attach_default_interface: ## Attach default libvirt network to CRC
+	-CURRENT_MAC=$(shell sudo virsh net-dumpxml default | grep crc | sed -e "s/.*mac='\(.*\)' name.*/\1/"); \
```

Comment:
I don't see how this finds `crc` in the default network definition.
```
gibi@bedrock:~                                                                                                                                                                                      $ sudo virsh net-dumpxml default
<network>
  <name>default</name>
  <uuid>d8d921ab-df9b-4626-b63c-cfdba21ffb07</uuid>
  <forward mode='nat'/>
  <bridge name='virbr0' stp='on' delay='0'/>
  <mac address='52:54:00:a8:34:cb'/>
  <ip address='192.168.122.1' netmask='255.255.255.0'>
    <dhcp>
      <range start='192.168.122.2' end='192.168.122.254'/>
    </dhcp>
  </ip>
</network>

gibi@bedrock:~                                                                                                                                                                                      $ sudo virsh net-dumpxml crc
<network connections='1'>
  <name>crc</name>
  <uuid>49eee855-d342-46c3-9ed3-b8d1758814cd</uuid>
  <forward mode='nat'>
    <nat>
      <port start='1024' end='65535'/>
    </nat>
  </forward>
  <bridge name='crc' stp='on' delay='0'/>
  <mac address='52:54:00:fd:be:d0'/>
  <ip family='ipv4' address='192.168.130.1' prefix='24'>
    <dhcp>
      <host mac='52:fd:fc:07:21:82' ip='192.168.130.11'/>
    </dhcp>
  </ip>
</network>

```

Reply:
is this output from after running this step? 
Getting the CURRENT_MAC is to do cleanup when rerunning this step to
* first detach and remove the entry from the network and don't fail if it does not exist using the '-' prefix
* second to attach a new interface with a new MAC address

This is to allow if someone reruns with a different CRC_DEFAULT_NETWORK_IP.

Reply:
I'll split this target to have a `crc_attach_default_interface_cleanup`. this aligns better with what we have in all the other make targets

Reply:
I guess I misunderstood the error messages coming from this make target. I try to follow the procedure you describe in the PR message and I see this locally after `crc start` finished successfully.

```console
$ make crc_attach_default_interface
CURRENT_MAC=; \
sudo virsh detach-interface crc network --mac "$CURRENT_MAC"
error: Failed to get option 'mac': Option argument is empty
error: Failed to detach interface

make: [Makefile:49: crc_attach_default_interface] Error 1 (ignored)
sudo virsh net-update default delete ip-dhcp-host "<host name='crc'/>" --config --live
error: Failed to update network default
error: Requested operation is not valid: network is not running

make: [Makefile:50: crc_attach_default_interface] Error 1 (ignored)
sleep 5
MAC_ADDRESS=52:54:00:AD:A8:F5; \
sudo virsh net-update default add-last ip-dhcp-host --xml "<host mac='$MAC_ADDRESS' name='crc' ip='192.168.122.10'/>" --config --live; \
sudo virsh attach-interface crc --source default --type network --model virtio --mac $MAC_ADDRESS --config --persistent; \
sleep 10; \
WORKER=crc-rwwzd-master-0; \
oc debug node/$WORKER -- ip -o link | awk 'toupper($0) ~ /ETHER $MAC_ADDRESS/{print $2}'  | awk -F: '{print $1}'
error: Failed to update network default
error: Requested operation is not valid: network is not running

error: Failed to attach interface
error: Requested operation is not valid: network 'default' is not active

Temporary namespace openshift-debug-9v68k is created for debugging node...
Starting pod/crc-rwwzd-master-0-debug ...
To use host binaries, run `chroot /host`

Removing debug pod ...
Temporary namespace openshift-debug-9v68k was removed.
```

Maybe it is more about the status of the `default` network on my machine

Reply:
```
$ sudo virsh net-list --all
 Name      State      Autostart   Persistent
----------------------------------------------
 crc       active     yes         yes
 default   inactive   no          yes

```

Reply:
``` console
$ sudo virsh net-start default
Network default started
```

```console
$ make crc_attach_default_interface
CURRENT_MAC=; \
sudo virsh detach-interface crc network --mac "$CURRENT_MAC"
error: Failed to get option 'mac': Option argument is empty
error: Failed to detach interface

make: [Makefile:49: crc_attach_default_interface] Error 1 (ignored)
sudo virsh net-update default delete ip-dhcp-host "<host name='crc'/>" --config --live
error: Failed to update network default
error: Requested operation is not valid: couldn't locate a matching dhcp host entry in network 'default'

make: [Makefile:50: crc_attach_default_interface] Error 1 (ignored)
sleep 5
MAC_ADDRESS=52:54:00:8B:3B:5A; \
sudo virsh net-update default add-last ip-dhcp-host --xml "<host mac='$MAC_ADDRESS' name='crc' ip='192.168.122.10'/>" --config --live; \
sudo virsh attach-interface crc --source default --type network --model virtio --mac $MAC_ADDRESS --config --persistent; \
sleep 10; \
WORKER=crc-rwwzd-master-0; \
oc debug node/$WORKER -- ip -o link | awk 'toupper($0) ~ /ETHER $MAC_ADDRESS/{print $2}'  | awk -F: '{print $1}'
Updated network default persistent config and live state
Interface attached successfully

Temporary namespace openshift-debug-lrr6s is created for debugging node...
Starting pod/crc-rwwzd-master-0-debug ...
To use host binaries, run `chroot /host`

Removing debug pod ...
Temporary namespace openshift-debug-lrr6s was removed.
```

Reply:
OK, it seems at least the interface get added to the crc domain and got an ip:
```
$ sudo virsh domifaddr crc
 Name       MAC address          Protocol     Address
-------------------------------------------------------------------------------
 vnet0      52:fd:fc:07:21:82    ipv4         192.168.130.11/24
 vnet1      52:54:00:8b:3b:5a    ipv4         192.168.122.10/24
```